ktor-utils/posix/src/io/ktor/util/CryptoNative.kt

```diff
@@ -4,12 +4,31 @@
 
 package io.ktor.util
 
+import kotlinx.cinterop.*
+import platform.Foundation.*
+import platform.posix.*
+
+private const val NONCE_SIZE_IN_BYTES = 16
 
 /**
- * Generates a nonce string. Could block if the system's entropy source is empty
+ * Generates a nonce string 16 characters long. Could block if the system's entropy source is empty
  */
 @InternalAPI
-actual fun generateNonce(): String = error("[generateNonce] is not supported on iOS")
+actual fun generateNonce(): String {
+    return hex(generateRandomByteArray())
+}
+
+private fun generateRandomByteArray(): ByteArray {
+    val result = ByteArray(NONCE_SIZE_IN_BYTES)
+    for (i in 0..3) {
+        val random = arc4random().toLong()
```

Comment:
Is `arc4random()` still cryptographically nowadays?

Reply:
So far I can only find the following online: https://stackoverflow.com/questions/44751523/is-there-a-cryptographically-secure-alternative-to-arc4random-on-ios 

An alternative could be https://developer.apple.com/documentation/security/1399291-secrandomcopybytes